### ISSUE TITLE
Make turtlebot_kinect_arm_calibration generic

### ISSUE THREAD

**Issue by shaun-edwards** [turtlebot/turtlebot_arm#1]
The turtlebot kinect to arm calibration package is useful for many different camera robot combinations.  It would be nice if this package could be made more generic (i.e. remove turtlebot references and update wiki to describe application of this tool to any robot).

**Comment by mikeferguson** [comment]
Just a note, the original code which this was based on can be found in the vanadium-ros-pkg qnd_calibration (https://code.google.com/p/vanadium-ros-pkg/source/browse/#svn%2Ftrunk%2Fsandbox%2Fqnd_calibration). I'm not sure how much stuff was added into the turtlebot code, but the original code might serve as some reference.

**Comment by corot** [comment]
@mikeferguson, I have a question marginally related with this issue:   I have updated the block manipulation demo for indigo/moveit, but I have some problems with the precision of blocks location, I imagine related to the camera calibration. My question is: the calibration process provides the tf from the robot to the camera_link, but uses the RGB images (referenced with rgb_optical_frame) as input. So I assumed that the calibration must do some internal frame conversions, but I cannot find such a thing within the code! So how the calibration "knows" where the camera_link must be if it only uses RGB images (with a different reference)? I wonder if my imprecision problems come from the fact that kinect internal frames on fuerte where different from the current ones provided on turtlebot_description kinect.urdf.xacro file.
Thanks!

PS: btw, the calibration and block manipulation work fine if I use my onboard Creative Senz3D!!!

**Comment by mikeferguson** [comment]
@corot It is possible the link name is wrong -- this code is >3 years old, and was unused for at least 1-2 years until recently, further it pre-date openni_launch being split into rgbd_launch, so it is possible that frame name defaults have changed.

**Comment by corot** [comment]
Right, you need to add an intermediate tf to deal with the new frames structure of the kinect. Added that, it works quite well, so I'll release with a warning in the wiki explaining what to do. Thanks!

**Comment by mikeferguson** [comment]
There are numerous alternatives for calibration at this point
